src/main/java/com/streamr/client/StreamrClient.java

```diff
@@ -447,7 +447,14 @@ public void publish(Stream stream, Map<String, Object> payload, Date timestamp,
 
     public void publish(Stream stream, Map<String, Object> payload, Date timestamp, String partitionKey, UnencryptedGroupKey newGroupKey) {
         if (!getState().equals(ReadyState.OPEN)) {
-            connect();
+            if (!userWantsToConnect) {
```

Comment:
**[minor]** This reads a bit funny due to current variable naming: "if the user doesn't want to connect, then connect!". I had to search through the rest of the file to figure out what `userWantsToConnect` actually is. 

However, I couldn't immediately come up with a clearly better name for the variable. A couple of (imperfect) suggestions that came to mind:
- `stayingConnected`
- `connectionLoopStarted`

Alternatively just a comment along the lines of "call connect() if the client isn't already attempting to stay connected" might do the trick.

Reply:
So the thinking here is similar to that of `autoConnect = true` in the streamr-client-javascript library: a convenience feature for the user to not have to invoke `connect` before `publish`. But yeah I definitely see how the naming is quite confusing here.  I'm leaning on renaming to `stayingConnected` _and_ adding a clarifying comment.